src/audio/pipeline.c

```diff
@@ -204,13 +204,17 @@ static void pipeline_trigger_sched_comp(struct pipeline *p,
 	case COMP_TRIGGER_PAUSE:
 	case COMP_TRIGGER_STOP:
 		pipeline_schedule_cancel(p);
+		p->status = COMP_STATE_PAUSED;
 		break;
 	case COMP_TRIGGER_START:
 		p->xrun_bytes = 0;
 
 		/* playback pipelines need scheduled now, capture pipelines are
-		 * scheduled once their initial DMA period is filled by the DAI */
-		if (comp->params.direction == SOF_IPC_STREAM_PLAYBACK) {
+		 * scheduled once their initial DMA period is filled by the DAI
+		 * or in resume process
+		 */
+		if (comp->params.direction == SOF_IPC_STREAM_PLAYBACK ||
+		    p->status == COMP_STATE_PAUSED) {
```

Comment:
It seems this will make 
`	case COMP_TRIGGER_START:`
and
`	case COMP_TRIGGER_RELEASE: `
The same now

Reply:
Yes, changed it already.

Reply:
this will have problem on BYT platform.
on BYT platform, it does not require this schedule.
my PR byt:pipeline:fix bug:panic when pause/release capture when play. #166 is trying to work for this.
you should refine this PR.

Reply:
Actually, when the stop command, if we want to follow the idea "reset the whole pipeline". this PR should not be merged. because the status of this pipeline is reset already before start comes.
1. for capture, no matter what kind of stop/start, the pipeline should start from initial state.
2. for capture, for pause/release, the pipeline should be scheduled. the old code does this job.
3. for playback, no matter what kind of stop/start, the pipeline should be start from initial state.
4. for playback, for pause/release, the pipeline should be scheduled. the old code does this job.

Reply:
Maybe I have to revet the commit 58bc284 to the test.

Reply:
It depends what is the requirement from the ALSA for the stop/start sequence.